src/google-maps/map-event-manager.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Observable, Subscriber} from 'rxjs';
+
+type MapEventManagerTarget = {
+  addListener: (name: string, callback: (...args: any[]) => void) => google.maps.MapsEventListener;
+} | undefined;
+
+/** Manages event on a Google Maps object, ensuring that events are added only when necessary. */
+export class MapEventManager {
+  /** Pending listeners that were added before the target was set. */
+  private _pending: {observable: Observable<any>, observer: Subscriber<any>}[] = [];
+  private _listeners: google.maps.MapsEventListener[] = [];
+  private _target: MapEventManagerTarget;
+
+  /** Clears all currently-registered event listeners. */
+  private _clearListeners() {
+    for (let listener of this._listeners) {
+      listener.remove();
+    }
+
+    this._listeners = [];
+  }
+
+  /** Gets an observable that add an event listener to the map when a consumer subscribes to it. */
+  getObservable<T>(name: string): Observable<T> {
```

Comment:
I think the name `getObservable` is a bit too generic. I could see any name like (get|create)Lazy(emitter|eventStream) (e.g. `getLazyEmitter` or `createLazyEventStream`)